cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -579,6 +579,11 @@ ParquetFileFragment::ParquetFileFragment(FileSource source,
       parquet_format_(checked_cast<ParquetFileFormat&>(*format_)),
       row_groups_(std::move(row_groups)) {}
 
+void ParquetFileFragment::set_bounds(int64_t start, int64_t end) {
+  ARROW_LOG(WARNING) << "Setting byte bounds for a ParquetFileFragment not supported! "
+                        "Use subset instead.";
```

Comment:
If we want to convey an error, we should not emit a warning message but return an error status.
Conversely, if we mean this to actually succeed (and be a no-op) on Parquet file fragments, then the warning message is merely distracting.